clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -672,6 +672,146 @@ check_function_markers()
         if (!run_checker(memrefs, false))
             return false;
     }
+    // Correctly skip return address check when the return address is
+    // unavailable.
+    {
+        std::vector<memref_t> memrefs = {
+            gen_instr(TID, 1),
+            gen_instr_type(TRACE_TYPE_INSTR_DIRECT_JUMP, TID, CALL_PC, CALL_SZ),
+            gen_marker(TID, TRACE_MARKER_TYPE_FUNC_ID, 2),
+            gen_marker(TID, TRACE_MARKER_TYPE_FUNC_RETADDR, /*pc=*/123456),
+        };
+        if (!run_checker(memrefs, false))
+            return false;
+    }
+    // Correctly handle nested function calls including tailcalls.
+    {
+        instr_t *label1 = INSTR_CREATE_label(GLOBAL_DCONTEXT);
+        instr_t *call1 = XINST_CREATE_call(GLOBAL_DCONTEXT, opnd_create_instr(label1));
```

Comment:
No reason to have the labels here right?  These can directly target the real instrs.

Reply:
I rewrote the code to remove encoding, and the labels.